examples/helia-browser-react-verified-fetch/src/App.tsx

```diff
@@ -1,25 +1,182 @@
-import { useCallback, useState } from 'react'
+// import { verifiedFetch, createVerifiedFetch } from '@helia/verified-fetch'
 import { verifiedFetch } from '@helia/verified-fetch'
+import { fileTypeFromBuffer } from '@sgtpooki/file-type'
+import { useCallback, useState } from 'react'
 import { helpText } from './constants'
 
-function App() {
+function renderOutput (output: string | JSX.Element, err: string): JSX.Element {
+  console.log('err: ', err)
+  console.log('output: ', output)
+  if (err.length > 0) {
+    return (
+      <div className="bg-red-300">
+        <pre className="bg-black text-red-300 rounded p-4">{err}</pre>
+      </div>
+    )
+  }
+
+  if (typeof output === 'string') {
+    return (
+      <div className="bg-violet-300">
+        {output != null && (
+          <pre className="bg-black text-teal-300 rounded p-4">
+            <code id="output" className="language-json">{`${output}`}</code>
+          </pre>
+        )}
+      </div>
+    )
+  }
+
+  return output
+}
+
+function loadingIndicator (message: string): JSX.Element {
+  return (
+    <div className="bg-yellow-300">
+      <pre className="bg-black text-yellow-300 rounded p-4">Loading... {message}</pre>
+    </div>
+  )
+}
+
+function App (): JSX.Element {
   const [path, setPath] = useState<string>('')
-  const [output, setOutput] = useState<string>('')
+  const [output, setOutput] = useState<string | JSX.Element>('')
   const [err, setErr] = useState<string>('')
+  const [loading, setLoadingTo] = useState<JSX.Element | null>(null)
+
+  const setSuccess = useCallback((message: string | JSX.Element) => {
+    setOutput(message)
+    setLoadingTo(null)
+    setErr('')
+  }, [])
+  const setError = useCallback((message: string) => {
+    setOutput('')
+    setLoadingTo(null)
+    setErr(message)
+  }, [])
+  const setLoading = useCallback((message: string) => {
+    setErr('')
+    setLoadingTo(loadingIndicator(message))
+  }, [])
```

Comment:
abstraction on top of the three potential displays that can happen. If you set loading, it should show that. If you set an error, it should show that. If you set output, it should show that.

Using these ensure the implicit rendering of content is handled appropriately, but we could use a `success`, `error`, and `loading` state and pass all three to `renderOutput`